src/openms_gui/source/VISUAL/Spectrum1DCanvas.cpp

```diff
@@ -940,8 +958,14 @@ namespace OpenMS
     if (peak.isValid())
     {
       QPoint begin;
-
-      const ExperimentType::PeakType& sel = getLayer_(layer_index).getCurrentSpectrum()[peak.peak];
+      const auto& spec = getLayer_(layer_index).getCurrentSpectrum();
+      if (peak.peak >= spec.size())
```

Comment:
I guess the reference can probably change by other signals. Can we somehow block those?

Reply:
since this is executed in the event loop until finished (the EL is not async), this is not possible IMHO